lib/mastodon/media_cli.rb

```diff
@@ -45,6 +45,91 @@ def remove
       say("Removed #{processed} media attachments (approx. #{number_to_human_size(aggregate)}) #{dry_run}", :green, true)
     end
 
+    option :days, type: :numeric, default: 60, aliases: [:d]
+    option :concurrency, type: :numeric, default: 5, aliases: [:c]
+    option :verbose, type: :boolean, default: false, aliases: [:v]
+    option :dry_run, type: :boolean, default: false
+    option :aggressive, type: :boolean, default: false, aliases: [:a]
+    desc 'prune-profile-media', 'Remove profile media files (headers, avatars)'
+    long_desc <<-DESC
+      Removes locally cached copies of profile media files from other servers.
+
+      The --days option specifies how old the last webfinger request to the user
+      has to be before they are removed. It defaults to 60 days.
+
+      If --aggressive is specified all non-local accounts will be pruned, by
+      default only accounts that are not followed by or following anyone locally
+      are pruned.
+    DESC
+    def prune_profile_media
+      time_ago   = options[:days].days.ago
+      dry_run    = options[:dry_run] ? '(DRY RUN)' : ''
+      aggressive = options[:aggressive]
+
+      processed, aggregate = parallelize_with_progress(
+        Account.where(last_webfingered_at: ..time_ago)
+               .left_outer_joins(:user)
+               .where(user: { id: nil })
+        ) do |account|
+        next if account.local?
+        next if (not account.avatar.present? and not account.header.present?)
+        next if not aggressive and Follow.where(account: account).or(Follow.where(target_account: account)).count > 0
```

Comment:
Normal convention in ruby would be to use `!` and `&&` instead of `not` and `and` (I've been a Ruby dev for 7 years and I actually thought `not` here was a syntax error, I didn't even realize it was valid syntax until I googled it lol).

```suggestion
        next if (!account.avatar.present? && !account.header.present?)
        next if !aggressive && Follow.where(account: account).or(Follow.where(target_account: account)).count > 0
```

Reply:
Ok that's probably the python speaking 🙂 will fix shortly